### PR TITLE
Automate virtlogd maxclients related cases

### PR DESCRIPTION
**Pull request by chunfuwen**
Automate virtlogd maxclients related cases

xx-59118,xx-86131,xx-86144,xx-86515

### CODE REVIEW

**Comment by chunfuwen** [comment]
DATA (filename=output.expected) => NOT FOUND (data sources: variant, test, file)
DATA (filename=stdout.expected) => NOT FOUND (data sources: variant, test, file)
DATA (filename=stderr.expected) => NOT FOUND (data sources: variant, test, file)
PASS 1-type_specific.io-github-autotest-libvirt.conf_file.qemu_conf.set_virtlogd.positive_test.valid_max_clients

DATA (filename=output.expected) => NOT FOUND (data sources: variant, test, file)
DATA (filename=stdout.expected) => NOT FOUND (data sources: variant, test, file)
DATA (filename=stderr.expected) => NOT FOUND (data sources: variant, test, file)
PASS 1-type_specific.io-github-autotest-libvirt.conf_file.qemu_conf.set_virtlogd.positive_test.valid_boundary_max_clients

DATA (filename=output.expected) => NOT FOUND (data sources: variant, test, file)
DATA (filename=stdout.expected) => NOT FOUND (data sources: variant, test, file)
DATA (filename=stderr.expected) => NOT FOUND (data sources: variant, test, file)
PASS 1-type_specific.io-github-autotest-libvirt.conf_file.qemu_conf.set_virtlogd.positive_test.nclients_occupied_released


Command 'virsh list' finished with 0 after 0.032307155s
cleaning libvirtd logs...
DATA (filename=output.expected) => NOT FOUND (data sources: variant, test, file)
DATA (filename=stdout.expected) => NOT FOUND (data sources: variant, test, file)
DATA (filename=stderr.expected) => NOT FOUND (data sources: variant, test, file)
PASS 1-type_specific.io-github-autotest-libvirt.conf_file.qemu_conf.set_virtlogd.negative_test.invalid_boundary_max_clients

File: libvirt/tests/src/daemon/conf_file/qemu_conf/set_virtlogd.py [hunk 1]
```diff
@@ -542,6 +556,317 @@ def check_record_save_restore_guest_log():
...
+                      % (nclients_msg, virtlogd_log_file))
+
+    def check_invalid_boundary_max_clients():
+        """
+        Check whether virtlogd can work normally when set invalid boundary max clients
+
+        """
+        virtlogd_config.max_clients = params.get("low_max_clients")
+        cmd = ("systemctl status virtlogd | grep 'Active: active'")
+        ret = process.run(cmd, ignore_status=True, shell=True, verbose=True)
```
Comment (nanli1) [inline libvirt/tests/src/daemon/conf_file/qemu_conf/set_virtlogd.py:635]: `ignore_status=True, shell=True` are using 9 times , could we set `virsh_dargs = {"ignore_status": True, "shell": True} in global
`

File: libvirt/tests/src/daemon/conf_file/qemu_conf/set_virtlogd.py [hunk 2]
```diff
@@ -654,6 +975,28 @@ def check_record_save_restore_guest_log():
...
+            elif expected_result == "log_filters_not_same_with_loglevel":
+                check_log_filters_not_same_with_loglevel()
+            elif expected_result == "reload_max_size_config":
+                check_reload_max_size_config()
+            elif expected_result == "set_max_size_config":
+                check_set_max_size_config()
+            elif expected_result == "set_max_backups":
+                check_set_max_backups()
+            elif expected_result == "reload_max_backups":
+                check_reload_max_backups()
```
Comment (nanli1) [inline libvirt/tests/src/daemon/conf_file/qemu_conf/set_virtlogd.py:999]: How about merge these lines?such as 
     ```
   if case in ["invalid_virtlogd_conf", "valid_max_clients",
                               "valid_boundary_max_clients", "nclients_occupied_released",
                               "invalid_boundary_max_clients", "reload_new_config",
                               "log_filters_not_same_with_loglevel", "reload_max_size_config",
                               "set_max_size_config",
                               "set_max_backups", "reload_max_backups"]:
```
```
check_test = eval("check_%s" % case)
check_test()
```

File: libvirt/tests/src/daemon/conf_file/qemu_conf/set_virtlogd.py (hunk 1 above)
Comment (chunfuwen) [reply libvirt/tests/src/daemon/conf_file/qemu_conf/set_virtlogd.py:635]: Define one global variable:    debug_param = {'ignore_status': True, 'shell':True, 'verbose':True}, and replace them with this variable

Comment (chunfuwen) [reply libvirt/tests/src/daemon/conf_file/qemu_conf/set_virtlogd.py:635]: Define one global variable: debug_param = {'ignore_status': True, 'shell':True, 'verbose':True}, and replace them with this variable

File: libvirt/tests/src/daemon/conf_file/qemu_conf/set_virtlogd.py (hunk 2 above)
Comment (chunfuwen) [reply libvirt/tests/src/daemon/conf_file/qemu_conf/set_virtlogd.py:999]: updated

**Review by nanli1 — APPROVED** [review]

**Review by dzhengfy — APPROVED** [review]
LGTM

**Pull request merged**